src/lib/transaction.ts

```diff
@@ -736,3 +456,327 @@ export async function sendTransaction(
+    ...sendOptions
+  } = options;
+
+  const hasComputeInstructions = instructions.some((ix) =>
+    ix.programId.equals(ComputeBudgetProgram.programId),
+  );
+
+  if (!hasComputeInstructions) {
+    const computeUnitBuffer = userComputeBuffer ?? { multiplier: 1.1 };
+    instructions = await addComputeInstructions(
+      connection,
+      instructions,
+      lookupTables ?? [],
+      signers[0].publicKey,
+      priorityFee,
+      computeUnitBuffer,
+      commitment,
+    );
+  }
+
+  const messageV0 = new TransactionMessage({
+    payerKey: signers[0].publicKey,
+    recentBlockhash: (await connection.getLatestBlockhash(commitment))
+      .blockhash,
+    instructions,
+  }).compileToV0Message(lookupTables);
+
+  const transaction = new VersionedTransaction(messageV0);
+
+  transaction.sign(signers);
+
+  return await sendVersionedTransactionWithRetry(
+    connection,
+    transaction,
+    sendOptions,
+  );
+}
+
+/**
+ * Adds compute unit price and limit instructions to the transaction
+ *
+ * @param connection - The Solana connection object
+ * @param instructions - Array of instructions to which compute unit instructions will be added
+ * @param lookupTables - Optional array of address lookup tables for account compression
+ * @param payer - The public key of the transaction payer
+ * @param priorityFee - Priority fee in microLamports (default: 10000 which is the minimum required for helius to see a transaction as priority)
+ * @param computeUnitBuffer - Optional buffer to add to simulated compute units
+ * @param commitment - Desired commitment level for the transaction
+ * @returns Array of instructions with compute unit instructions added
+ */
+async function addComputeInstructions(
+  connection: Connection,
+  instructions: Array<TransactionInstruction>,
+  lookupTables: Array<AddressLookupTableAccount>,
+  payer: PublicKey,
+  priorityFee: number = 10000,
+  computeUnitBuffer: ComputeUnitBuffer = {},
+  commitment: Commitment = "confirmed",
+): Promise<Array<TransactionInstruction>> {
+  instructions.push(
+    ComputeBudgetProgram.setComputeUnitPrice({
+      microLamports: priorityFee,
+    }),
+  );
+
+  const simulatedCompute = await getSimulationComputeUnits(
+    connection,
+    instructions,
+    payer,
+    lookupTables,
+    commitment,
+  );
+
+  if (simulatedCompute === null) {
+    throw new Error("Failed to simulate compute units");
+  }
+
+  console.log("Simulated compute units", simulatedCompute);
+
+  // Apply buffer to compute units
+  let finalComputeUnits = simulatedCompute;
+  if (computeUnitBuffer.multiplier) {
+    finalComputeUnits = Math.floor(
+      finalComputeUnits * computeUnitBuffer.multiplier,
+    );
+  }
+  if (computeUnitBuffer.fixed) {
+    finalComputeUnits += computeUnitBuffer.fixed;
+  }
+
+  console.log("Final compute units (with buffer)", finalComputeUnits);
+
+  instructions.push(
+    ComputeBudgetProgram.setComputeUnitLimit({
+      units: finalComputeUnits,
+    }),
+  );
```

Comment:
similar CU bug as before, but worse. 

if they already have a `setComputeUnitLimit` instruction in the transaction, it will now have two of the same compute budget instructions. therefore the transaction will always fail